mlrun/datastore/sources.py

```diff
@@ -49,6 +51,53 @@ def get_source_step(source, key_fields=None, time_field=None, context=None):
     return source.to_step(key_fields, time_field, context)
 
 
+class _MondoDBIterator:
+    def __init__(self, collection, iter_chunksize, iter_query):
+        """
+        Iterate over given mongoDB collection
+
+        :param iter_query:       dictionary query for mongodb
+        :param iter_chunksize:   number of rows per chunk
+        :param collection:  a mongodb collection
+        """
+        self.my_collection_iter = collection.find_raw_batches(
+            iter_query, batch_size=iter_chunksize
+        )
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        batch = next(self.my_collection_iter)
+        decode_batch = bson.decode_all(batch)
+        curr_df = pd.DataFrame(decode_batch)
```

Comment:
I think keeping the same data in 3 different variables may be inefficient as you rely on the garbage collector to free them. Maybe doing:
```
curr_df = pd.DataFrame(bson.decode_all(next(self.my_collection_iter)))
``` 
can reduce this friction (I'm not sure though).
How frequent will this be called and should it happen fast?